rust/sedona-schema/src/datatypes.rs

```diff
@@ -333,6 +365,1241 @@ fn deserialize_edges(edges: &Value) -> Result<Edges> {
+                    .value(raster_index),
+                min_y: self
+                    .bbox
+                    .column(bounding_box_indices::MIN_Y)
+                    .as_any()
+                    .downcast_ref::<Float64Array>()
+                    .unwrap()
+                    .value(raster_index),
+                max_x: self
+                    .bbox
+                    .column(bounding_box_indices::MAX_X)
+                    .as_any()
+                    .downcast_ref::<Float64Array>()
+                    .unwrap()
+                    .value(raster_index),
+                max_y: self
+                    .bbox
+                    .column(bounding_box_indices::MAX_Y)
+                    .as_any()
+                    .downcast_ref::<Float64Array>()
+                    .unwrap()
+                    .value(raster_index),
+            })
+        }
+    }
+}
+
+impl<'a> RasterRef for RasterRefImpl<'a> {
+    fn metadata(&self) -> &dyn MetadataRef {
+        &self.metadata
+    }
+
+    fn crs(&self) -> Option<&str> {
+        if self.crs.is_null(self.bands.raster_index) {
+            None
+        } else {
+            Some(&self.crs.value(self.bands.raster_index))
+        }
+    }
+
+    fn bands(&self) -> &dyn BandsRef {
+        &self.bands
+    }
+}
+
+/// Iterator over raster structs in an Arrow StructArray
+///
+/// This provides efficient, zero-copy access to raster data stored in Arrow format.
+/// Each iteration yields a `RasterRefImpl` that provides access to both metadata and band data.
+pub struct RasterStructIterator<'a> {
+    raster_array: &'a StructArray,
+    current_row: usize,
+}
+
+impl<'a> RasterStructIterator<'a> {
+    /// Create a new iterator over the raster struct array
+    pub fn new(raster_array: &'a StructArray) -> Self {
+        Self {
+            raster_array,
+            current_row: 0,
+        }
+    }
+
+    /// Get the total number of rasters in the array
+    pub fn len(&self) -> usize {
+        self.raster_array.len()
+    }
+
+    /// Check if the array is empty
+    pub fn is_empty(&self) -> bool {
+        self.raster_array.is_empty()
+    }
+
+    /// Get a specific raster by index without consuming the iterator
+    pub fn get(&self, index: usize) -> Option<RasterRefImpl<'a>> {
+        if index >= self.raster_array.len() {
+            return None;
+        }
+
+        Some(RasterRefImpl::new(self.raster_array, index))
+    }
+}
+
+impl<'a> Iterator for RasterStructIterator<'a> {
+    type Item = RasterRefImpl<'a>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.current_row < self.raster_array.len() {
+            let result = self.get(self.current_row)?;
+            self.current_row += 1;
+            Some(result)
+        } else {
+            None
+        }
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let remaining = self.raster_array.len().saturating_sub(self.current_row);
+        (remaining, Some(remaining))
+    }
+}
```

Comment:
I'm uncertain should we implement `Iterator` trait for `RasterStructIterator`, since the only actual usage of this type is random access (`get(&self, index: usize)`).